dm_control/locomotion/walkers/assets/dog_v2/utils.py

```diff
@@ -0,0 +1,213 @@
+        out_file_name = mjcf_model.model + '.xml'
+    elif not out_file_name.lower().endswith('.xml'):
+        raise ValueError('If `out_file_name` is specified it must end with '
+                         '\'.xml\': got {}'.format(out_file_name))
+
+    assets = mjcf_model.get_assets()
+    new_dict = {}
+    for k, v in assets.items():
+        new_k = save_meshes + k
+        new_dict[new_k] = v
+
+    xml_string = mjcf_model.to_xml_string()
+    index = xml_string.find('compiler') + len("compiler")
+    if bool(meshdir):
+        xml_string = xml_string[:index] + " meshdir=\"" + \
+            meshdir + '\" ' + xml_string[index:]
+
+    xml_string = xml_string.replace('general', 'muscle')
+
+    # remove extra default class='/'
+    start = xml_string.find('<default')
+    end = xml_string.find('</default>')
+    xml_string = xml_string.replace(xml_string[start: end + len('</default>')], '')
+
+    new_dict[out_dir + out_file_name] = xml_string
+
+    if not os.path.exists(out_dir):
+        os.makedirs(out_dir)
+
+    for path, contents in new_dict.items():
+        with open(path, 'wb') as f:
+            f.write(util.to_binary_string(contents))
+
+
+def slices2paths(mtu, slices, muscle_length):
+    paths = []
+    area = -1
+    for cross_section in slices:
+        if cross_section is not None:
+            try:
+                tmp_area = cross_section.area
+            except Exception as e:
+                tmp_area = -1
+            
+            if tmp_area > area:
+                area = tmp_area
+
+            path = cross_section.to_3D()
+
+            v = np.array(path.vertices)
+            x_min = np.min(v[:, 0])
+            x_max = np.max(v[:, 0])
+
+            z_min = np.min(v[:, 2])
+            z_max = np.max(v[:, 2])
+
+            y_min = np.min(v[:, 1])
+            y_max = np.max(v[:, 1])
+
+            if mtu in LATERAL:
+                test = [path.centroid,
+                np.array([path.centroid[0], path.centroid[1], z_min]),
+                np.array([path.centroid[0], path.centroid[1], z_max]),
+                np.array([path.centroid[0], y_max, path.centroid[2]]),
+                np.array([path.centroid[0], y_min, path.centroid[2]])]
+            else:
+                test = [path.centroid,
+                np.array([x_max, path.centroid[1], path.centroid[2]]),
+                np.array([x_min, path.centroid[1], path.centroid[2]]),
+                np.array([path.centroid[0], y_max, path.centroid[2]]),
+                np.array([path.centroid[0], y_min, path.centroid[2]])]
+            paths.append(test)
+
+    const = round(muscle_length * 100)
+    if len(paths) > const:
+        indices = np.linspace(start=0,
+            stop=len(paths) - 1,
+            num=const,
+            dtype=int)
+
+    tmp = []
+    for idx in indices:
+        tmp.append(paths[idx])
+
+    paths = tmp
+    assert area != -1
+    return paths, area
+
+
+def getClosestGeom(kd_tree, point, mjcf, mtu):
```

Comment:
get_closest_geom

Is this a duplicate of the other one? Maybe extract to a common file.